chatsky/processing/standard.py

```
@@ -39,3 +46,45 @@ async def call(self, ctx: Context) -> MessageInitTypes:
                 return await processing_object.modified_response(current_response, ctx)
 
         ctx.current_node.response = ModifiedResponse()
+
+
+class AddFallbackResponses(ModifyResponse, arbitrary_types_allowed=True):
+    """
+    ModifyResponse with pre-response processing to handle exceptions dynamically.
+    """
+
+    exception_responses: Dict[Union[Type[Exception], Literal["Else"]], AnyResponse]
+    """
+    Dictionary mapping exception types to fallback responses.
+    """
+
+    @field_validator("exception_responses")
+    @classmethod
+    def validate_not_empty(cls, exception_responses: dict) -> dict:
+        """
+        Validate that the `exception_responses` dictionary is not empty.
+
+        :param exception_responses: Dictionary mapping exception types to fallback responses.
+        :raises ValueError: If the `exception_responses` dictionary is empty.
+        :return: Not empty dictionary of exception_responses.
+        """
+        if len(exception_responses) == 0:
+            raise ValueError("Exceptions dict is empty")
+        return exception_responses
+
+    async def modified_response(self, original_response: BaseResponse, ctx: Context) -> MessageInitTypes:
+        """
+        Catch response errors and process them based on `exception_responses`.
+
+        :param original_response: The original response of the current node.
+        :param ctx: The current context.
+
+        :return: Message to replace original response with modified due to fallback response.
+        """
+        result = await original_response.wrapped_call(ctx)
+        if isinstance(result, Exception):
+            exception = self.exception_responses.get(type(result), self.exception_responses.get("Else"))
```

This will return `None` if "Else" is not in `exception_responses`, leading to an exception.

I think that instead we should re-raise an existing exception:
```python
framework_data.response_exception = str(result)
if exception is None:
    raise result
```

---

747c34aad69762f1f224b3731a59d7ad0c3f92ff